### PR TITLE
handle interactions w/ explosions

### PR DESCRIPTION
**Pull request by fluxionary**
this was originally https://github.com/pyrollo/display_modpack/pull/50

1. sign entities shouldn't be destroyed by explosions if the sign isn't destroyed
2. sign entities (and node metadata) should be removed if the sign is destroyed by an explosion.

### CODE REVIEW

File: display_api/display.lua [hunk 1]
```diff
@@ -251,6 +251,15 @@ function display_api.on_destruct(pos)
 	end
 end
 
+function display_api.on_blast(pos, intensity)
+	if not minetest.is_protected(pos, "tnt:blast") then
```
Comment (wsor4035) [inline display_api/display.lua]: from the docs
`name will be "" for non-players or unknown players.`
probably shouldnt be passing a node name style string in

File: display_api/display.lua [hunk 2]
```diff
@@ -269,13 +278,21 @@ end
 function display_api.register_display_entity(entity_name)
 	if not minetest.registered_entities[entity_name] then
 		minetest.register_entity(':'..entity_name, {
-			collisionbox = { 0, 0, 0, 0, 0, 0 },
-			visual = "upright_sprite",
-			textures = {},
+			initial_properties = {
+				collisionbox = {0, 0, 0, 0, 0, 0},
+				visual = "upright_sprite",
+				textures = {},
```
Comment (wsor4035) [inline display_api/display.lua:284]: is this needed?

File: display_api/display.lua [hunk 3]
```diff
@@ -269,13 +278,21 @@ end
...
+				physical = false,
```
Comment (wsor4035) [inline display_api/display.lua]: is this needed? default according to the docs

**Review by wsor4035 — COMMENTED** [review]
some initial thoughts

File: display_api/display.lua (hunk 3 above)
Comment (wsor4035) [reply display_api/display.lua]: hmm, not sure why github turned one line into 4, refering to `physical = false,`

**Comment by fluxionary** [comment]
i'll get back to this at some point, it's on the todo list

**Comment by Niklp09** [comment]
@fluxionary Can we merge this?

**Review by BuckarooBanzay — APPROVED** [review]

**Pull request merged**